pinot-query-planner/src/main/java/org/apache/pinot/query/planner/StageMetadata.java

```diff
@@ -81,6 +82,12 @@ public void attach(StageNode stageNode) {
       _requiresSingletonInstance = _requiresSingletonInstance || (sortNode.getCollationKeys().size() > 0
           && sortNode.getOffset() != -1);
     }
+    if (stageNode instanceof WindowNode) {
+      WindowNode windowNode = (WindowNode) stageNode;
+      // Empty OVER() and OVER(ORDER BY) need to be processed on a singleton node. OVER() with PARTITION BY can be
```

Comment:
> Empty OVER() and OVER(ORDER BY) need to be processed on a singleton node.

Add a TODO to improve this in future per original discussion ?

Reply:
done